src/core/blockchain/wallet.cr

```diff
@@ -1,112 +1,82 @@
-      secret_key:   String,
-      public_key_x: String,
-      public_key_y: String,
-      address:      String,
+      private_key: String,
+      public_key: String,
+      wif: String,
+      address: String,
     })
 
-    getter secret_key : String
-    getter public_key_x : String
-    getter public_key_y : String
+    getter private_key : String
+    getter public_key : String
+    getter wif : String
     getter address : String
 
-    def initialize(@secret_key : String, @public_key_x : String, @public_key_y : String, @address : String)
+    def initialize(@private_key : String, @public_key : String, @wif : String, @address : String)
     end
 
     def verify!
-      Wallet.verify!(@secret_key, @public_key_x, @public_key_y, @address)
+      Wallet.verify!(@private_key, @public_key, @wif, @address)
     end
 
-    def self.from_path(wallet_path : String) : self
+    def self.from_path(wallet_path : String) : EncryptedWallet
       raise "Failed to find wallet at #{wallet_path}, create it first!" unless File.exists?(wallet_path)
 
-      self.from_json(File.read(wallet_path))
-    end
-
-    def self.create(testnet = false)
-      key_pair = create_key_pair
-      address = public_key_to_address(key_pair[:public_key], testnet)
-
-      {
-        secret_key:   Base64.strict_encode(key_pair[:secret_key].to_s(base: 10)),
-        public_key_x: Base64.strict_encode(key_pair[:public_key].x.to_s(base: 10)),
-        public_key_y: Base64.strict_encode(key_pair[:public_key].y.to_s(base: 10)),
-        address:      address,
-      }
+      EncryptedWallet.from_json(File.read(wallet_path))
     end
 
-    def self.create_key_pair
-      secp256k1 = ECDSA::Secp256k1.new
-      key_pair = secp256k1.create_key_pair
+    def self.decrypt(encrypted_wallet : EncryptedWallet, password) : Wallet
+      begin
+        decrypted_wallet = BlowFish.decrypt(encrypted_wallet.cipher, password)
+      rescue ex
+        raise "Failed to decrypt wallet with supplied password"
+      end
 
-      {
-        secret_key: key_pair[:secret_key],
-        public_key: key_pair[:public_key],
-      }
+      wallet = Wallet.from_json(decrypted_wallet)
+      verify!(wallet.private_key, wallet.public_key, wallet.wif, wallet.address)
+      wallet
     end
 
-    def self.verify!(secret_key : String,
-                     public_key_x : String,
-                     public_key_y : String,
-                     address : String) : Bool
-      secp256k1 = ECDSA::Secp256k1.new
-      secret_key_raw = BigInt.new(Base64.decode_string(secret_key), base: 10)
-      public_key_raw_x = BigInt.new(Base64.decode_string(public_key_x), base: 10)
-      public_key_raw_y = BigInt.new(Base64.decode_string(public_key_y), base: 10)
+    def self.create(password ,testnet = false) : EncryptedWallet
 
-      raise "Invalid checksum for #{address}" unless valid_checksum?(address)
+      raise "Password is too short, it must be a minimum of 16 characters" if password.size < 16
```

Comment:
16 is too long for a usability I think. 8 should be. Otherwise I should **recommend** (not must be) to include symbolic-characters and numbers in the password when user create a wallet.